logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/slow/SlowSyncManager.kt

```diff
@@ -105,8 +108,15 @@ internal class SlowSyncManager(
                 logger.i("Next SlowSync should be performed on '$nextSlowSyncDateTime'")
                 val lastVersion = slowSyncRepository.getSlowSyncVersion()
                 logger.i("Last saved SlowSync version is $lastVersion, current is $CURRENT_VERSION")
-                currentTime > nextSlowSyncDateTime || CURRENT_VERSION > lastVersion
-            } ?: true
+
+                if (currentTime > nextSlowSyncDateTime) {
+                    SlowSyncParam.LastSlowSyncTooOld
+                } else if (CURRENT_VERSION > lastVersion) {
+                    SlowSyncParam.MigrationNeeded(oldVersion = lastVersion, newVersion = CURRENT_VERSION)
```

Comment:
What if `curentTime > nextSlowSyncDateTime` but migration is also needed?

I guess the first condition should be the migration, as it will also force a slow sync, just like when the last was too old.

Reply:
good catch i will move the check for MigrationNeeded before the LastSlowSyncTooOld and add tests for it